client/src/featureform/register.py

```diff
@@ -1967,25 +1970,47 @@ def register_column_resources(
 
     def __get_feature_nv(self, features):
         feature_nv_list = []
+        feature_lags = []
         for feature in features:
             if isinstance(feature, str):
                 feature_nv = (feature, "default")
                 feature_nv_list.append(feature_nv)
             elif isinstance(feature, dict):
-                feature_nv = (feature["name"], feature["variant"])
-                feature_nv_list.append(feature_nv)
+                lag = feature.get("lag")
+                if lag:
+                    if set(["lag", "feature", "variant"]) in set(feature.keys()):
+                        raise ValueError(f"feature lags require 'lag', 'feature', 'variant' fields. Received: {feature.keys()}")
+                    
+                    if not isinstance(lag, timedelta):
+                        raise ValueError(f"the lag, '{lag}', needs to be of type 'datetime.timedelta'. Received: {type(lag)}.")
+
+                    feature_name_variant = (feature["feature"], feature["variant"])
+                    if feature_name_variant not in feature_nv_list:
+                        feature_nv_list.append(feature_name_variant)
+
+                    lag_name = f"{feature['feature']}_{feature['variant']}_lag_{lag}"
+                    sanitized_lag_name = lag_name.replace(" ", "").replace(",", "_").replace(":", "_")
+                    feature["name"] = feature.get("name", sanitized_lag_name)
+                    
+                    feature_lags.append(feature)
+                else:
+                    feature_nv = (feature["name"], feature["variant"])
+                    feature_nv_list.append(feature_nv)
             elif isinstance(feature, list):
                 feature_nv_list.extend(self.__get_feature_nv(feature))
             else:
                 feature_nv_list.append(feature)
-        return feature_nv_list
+
+        unique_feature_nv_list = list(set(feature_nv_list))
+        unique_feature_lags = list(set(feature_lags))
+        return unique_feature_nv_list, unique_feature_lags
```

Comment:
Is this actually needed? If we register feature lags as part of a training set at this point, they don't actually have any definition except their derivation from the original features. Could they just remain in the training set definition, untouched until they get to the coordiantor?